lib/rubocop/cop/rspec/receive_counts.rb

```diff
@@ -0,0 +1,82 @@
+module RuboCop
+  module Cop
+    module RSpec
+      # Check for `once` and `twice` receive counts matchers usage.
+      #
+      # @example
+      #
+      #     # bad
+      #     expect(foo).to receive(:bar).exactly(1).times
+      #     expect(foo).to receive(:bar).exactly(2).times
+      #     expect(foo).to receive(:bar).at_least(1).times
+      #     expect(foo).to receive(:bar).at_least(2).times
+      #     expect(foo).to receive(:bar).at_most(1).times
+      #     expect(foo).to receive(:bar).at_most(2).times
+      #
+      #     # good
+      #     expect(foo).to receive(:bar).once
+      #     expect(foo).to receive(:bar).twice
+      #     expect(foo).to receive(:bar).at_least(:once)
+      #     expect(foo).to receive(:bar).at_least(:twice)
+      #     expect(foo).to receive(:bar).at_most(:once)
+      #     expect(foo).to receive(:bar).at_most(:twice).times
+      #
+      class ReceiveCounts < Cop
+        include RangeHelp
+
+        MSG = 'Use `%<alternative>s` instead of `%<original>s`.'.freeze
+
+        def_node_matcher :receive_counts, <<-PATTERN
+          (send $(send _ {:exactly :at_least :at_most} (int ${1 2})) :times)
+        PATTERN
+
+        def on_send(node)
+          receive_counts(node) do |offending_node, count|
+            add_offense(
+              offending_node,
+              message: message_for(offending_node.method_name, count),
+              location: range(node, offending_node)
+            )
+          end
+        end
+
+        def autocorrect(node)
+          lambda do |corrector|
+            replacement = matcher_for(
+              node.method_name,
+              node.first_argument.source.to_i
+            )
+            corrector.replace(
+              range(node.parent, node),
+              replacement.sub('.', '') # remove the leading dot
+            )
+          end
+        end
+
+        private
+
+        def message_for(method, count)
+          alternative = matcher_for(method, count)
+          original = ".#{method}(#{count}).times"
```

Comment:
Shouldn’t this be available from the node source?

Reply:
The receiver of `exactly` is the `.receive` node (which may have other calls chained, like `with`). I'm trying to match `times` and only the selector part of `exactly`, which is a bit tricky, but makes autocorrection easy.
Having said that, it's possible to get substring from the original source, from the point of `times` call until the end

Reply:
Nah, this is probably fine.

Reply:
Although it would have to operate on source of node, while method is from offending_node. And generally using slice won't be much prettier

Reply:
But I can get the source from the range, so after all it's not a bad solution